src/meshapi/views/forms.py

```diff
@@ -140,8 +141,14 @@ def process_join_form(r: JoinFormRequest, request: Optional[Request] = None) ->
     if not r.email_address:
         return Response({"detail": "Must provide an email"}, status=status.HTTP_400_BAD_REQUEST)
 
-    if r.email_address and not validate_email_address(r.email_address):
-        return Response({"detail": f"{r.email_address} is not a valid email"}, status=status.HTTP_400_BAD_REQUEST)
+    try:
+        if r.email_address and not validate_email_address(r.email_address):
+            return Response({"detail": f"{r.email_address} is not a valid email"}, status=status.HTTP_400_BAD_REQUEST)
+    except EmailValidationError:
```

Comment:
Wait, this is a little confusing, but I think I understand what's happening. Are DNSTimeoutError, SMTPCommunicationError, and TLSNegotiationError extensions of EmailValidationError?

You might consider leaving a note here for the future to point this out, if indeed that's what's happening.

Reply:
You have it correct. Added a comment